client/src/components/Schedule/FormEntity.tsx

```diff
@@ -0,0 +1,333 @@
+import React, { useState } from 'react';
+import { Task, TaskService } from 'services/task';
+import { CourseEvent, CourseService } from 'services/course';
+import { withSession } from 'components';
+import { UserSearch } from 'components/UserSearch';
+import { UserService } from 'services/user';
+import { formatTimezoneToUTC } from 'services/formatter';
+import { union } from 'lodash';
+import { Form, Input, InputNumber, Button, DatePicker, Select, Alert, Row, Col } from 'antd';
+import moment from 'moment-timezone';
+import { EVENT_TYPES, SPECIAL_ENTITY_TAGS } from './model';
+import { TIMEZONES } from '../../configs/timezones';
+import { Event, EventService } from 'services/event';
+
+const { Option } = Select;
+
+const layout = {
+  labelCol: { span: 6 },
+  wrapperCol: { span: 18 },
+};
+
+const validateMessages = {
+  required: '${label} is required!',
+  types: {
+    email: '${label} is not validate email!',
+    number: '${label} is not a validate number!',
+  },
+  number: {
+    range: '${label} must be between ${min} and ${max}',
+  },
+  links: 'Input link or delete this field.',
+};
+
+type Props = {
+  handleCancel: () => void;
+  onFieldsChange: (values: any) => void;
+  courseId: number;
+  entityType: string;
+  onEntityTypeChange: (type: string) => void;
+  typesFromBase: string[];
+  editableRecord: CourseEvent | null;
+  refreshData: Function;
+};
+
+const FormAddEntity: React.FC<Props> = ({
+  handleCancel,
+  courseId,
+  typesFromBase,
+  onFieldsChange,
+  onEntityTypeChange,
+  entityType,
+  editableRecord,
+  refreshData,
+}) => {
+  const [isSuccess, setSuccess] = useState(false);
```

Comment:
`success` is a quite vague name... let's replace with something more specific... For example `isFormSubmitted`

Reply:
fix